lib/orphans.nix

```diff
@@ -0,0 +1,43 @@
+{ lib }:
+let
+  inherit (lib) types optional isDerivation;
+  inherit (lib.lists) flatten any;
+  inherit (lib.asserts) assertMsg;
+  inherit (builtins) hasAttr length concatStringsSep mapAttrs attrValues tryEval;
+
+  isAttr = types.attrs.check;
+in
+rec {
+  hasMaintainer = p:
```

Comment:
Can you please add some documentation on how to use this function?

Example: https://github.com/NixOS/nixpkgs/blob/master/lib/attrsets.nix#L15-L23

Reply:
I've decomposed that in a few more, added a bit of documentation for all.